File: Java Based Archives/ViewModel/app/src/main/java/com/android/viewmodel/MainActivity.java
package com.android.viewmodel;

// Import necessary classes
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.lifecycle.ViewModelProvider;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class MainActivity extends AppCompatActivity {
    MainActivityViewModel mainActivityViewModel;

    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // Set the layout for the activity
        setContentView(R.layout.activity_main);

        // Find and set up the Toolbar
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        // Initialize the ViewModel
        mainActivityViewModel = new ViewModelProvider(this).get(MainActivityViewModel.class);

        // Find the TextView for displaying count
        textView = findViewById(R.id.tvCount);
        // Set the initial count in the TextView
        textView.setText("Count is: " + mainActivityViewModel.getInitialCount());

        // Find and set up the FloatingActionButton
        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // Update the count and display in the TextView when FloatingActionButton is clicked
                textView.setText("Count is: " + mainActivityViewModel.getCurrentCount());
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here.
        int id = item.getItemId();

        if (id == R.id.action_settings) {
            // Handle settings action
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: ViewStateDemo/app/src/main/java/com/example/viewstatedemo/MainActivity.kt
package com.example.viewstatedemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    private var count = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewButton()
    }

    private fun viewButton() {
        val button = findViewById<Button>(R.id.btnCount)
        button.text = "Count is : $count"

        button.setOnClickListener {
            button.text = "Count is : ${++count}"
            Toast.makeText(
                this, "Count is : $count",
                Toast.LENGTH_SHORT
            ).show()
        }
    }


}